test/mqtt.js

```diff
@@ -40,6 +40,20 @@ describe('mqtt', function () {
       c.options.should.have.property('username', 'user')
     })
 
+    it('should return an MqttClient with the clientid with random value', function () {
+      var c = mqtt.connect('mqtt://user@localhost:1883')
+
+      c.should.be.instanceOf(mqtt.MqttClient)
+      c.options.should.have.property('clientId')
```

Comment:
Can you call `c.end()` everywhere?

Reply:
I'll add to every cases? I think except one.